From 91bac12965f98d88e8d98bf88356a51ee2dd0811 Mon Sep 17 00:00:00 2001
From: Rick Farrington <rfarrington@marvell.com>
Date: Thu, 23 Apr 2020 18:09:13 -0400
Subject: [PATCH 541/767] drivers: marvell: otx2-sdei-ghes: Set GHES
 block_status correctly.

commit c93cf9a71278170951cc7c8c3490dfafadea927f from
git@git.assembla.com:cavium/WindRiver.linux.git

The GHES block_status is only required to contain the count
of entries contained in the data section.
The error status bits are optional and only server to enhance
the Error Severity of each individual entry.
Therefore, this commit removes the hard-coded
'ACPI_HEST_CORRECTABLE' status as it is neither correct nor
required.

Change-Id: I3b2944cb82867438a263f344f479a99c2cbd3871
Signed-off-by: Rick Farrington <rfarrington@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27417
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
index cd2e33937270..e4a0a727fae2 100644
--- a/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
+++ b/drivers/soc/marvell/octeontx2-ghes/otx2-sdei-ghes.c
@@ -123,8 +123,11 @@ static int sdei_ghes_callback(u32 event_id, struct pt_regs *regs, void *arg)
 		 */
 		wmb();
 
-		estatus->block_status = ACPI_HEST_CORRECTABLE |
-					(1 << 4); /* i.e. one entry */
+		/*
+		 * This simply needs the entry count to be non-zero.
+		 * Set entry count to one (see ACPI_HEST_ERROR_ENTRY_COUNT).
+		 */
+		estatus->block_status = (1 << 4); /* i.e. one entry */
 
 		if (++tail >= event->ring->size)
 			tail = 0;
-- 
2.31.1

